examples/postgres/test_all

```diff
@@ -1,7 +1,22 @@
 #!/bin/sh
 set -e
 
-export DATABASE_URL="postgres://localhost/diesel_examples"
+set -a
+. ../../.env || true
+DATABASE_URL=${DATABASE_URL-${PG_DATABASE_URL}}
+set +a
+
+trap finish EXIT
+
+function finish {
+  # Make sure when we're done, we leave behind an empty database. Otherwise,
+  # at the time of writing, running diesel_tests stumbles as it tries to run
+  # conflicting migrations on an overlapping schema.
+  #
+  # TODO: Find a better way to avoid mixing of resources across tests.
+  ../../bin/diesel database drop
+  ../../bin/diesel database touch
```

Comment:
Removing this hard-coding was part of why I opened this PR (I thought that was clear from the description and the commit messages). The instructions in `CONTRIBUTING.md` suggest that you get to decide what you're running the tests against, and the hard-coding doesn't work for everyone (even with trivial local servers, for example I would have to add `postgres@` to the hardcoded string).

So, removing the hard-coding, I naturally let them run against what we have, which is `{MYSQL,PG}_DATABASE_URL`, but as you pointed out, that's not a good idea due to the interaction with the tests. That's why I remove that state and so it works, but I agree that it's not a good solution (for example, if the example fails half-way through).

One could argue that it would be easiest to just hardcode it all and force folks to run things at the right location. I don't think that's a great solution though, and certainly it isn't fun for CI, (for example, you don't run the example in appveyor, probably because there you require user and password) but it could work.

My preference so far (and one that should hopefully also reflect your position) is an extra set of `.env` vars to be used for running the examples.

When I cloned the project, I wasn't planning on spending so much time on running the tests, but now that I have I think there's some work to be done to lower the barrier of entry, and ideally that would mean opening `CONTRIBUTING.md`, starting the databases *anywhere I want*, creating an appropriate `.env` and then be able to run `./bin/test` (i.e. what I'm trying to achieve here).